responses/__init__.py

```diff
@@ -601,7 +620,35 @@ def add_passthru(self, prefix):
         """
         if not isinstance(prefix, Pattern) and _has_unicode(prefix):
             prefix = _clean_unicode(prefix)
-        self.passthru_prefixes += (prefix,)
+        self._passthru_prefixes += (prefix,)
+        response = PassthroughResponse(method, prefix)
+        response.check_fired = check_fired
+        response.priority = Priority.LOW
+        self.add(response)
+
+    @property
+    def passthru_prefixes(self):
+        warn(
+            "Direct access of passthru_prefixes is depricated. "
+            "Passthroughs are stored as PassthroughResponses.",
+            DeprecationWarning,
+        )
+        return self._passthru_prefixes
```

Comment:
Couldn't prefixes be generated at runtime from the pass through responses? I also don't understand why this method needs to be deprecated.

Reply:
I think this is a misunderstanding, I absolutely in favour of the method `add_passthru`, because after getting to know it better, I clearly see the benefit. In the current implementation, it will not through any warning, because it will only interact with the private attribute `_passthru_prefixes`. The desire to deprecate the public attribute/property `passthru_prefixes` (especially the setter) comes from thinking about PEP 20 (Zen of Python): "*There should be one-- and preferably only one --obvious way to do it.*". The method `add_passthru` is already the preferred setter method, so I would discourage to offer this alternative way and therefore remove it from the public interface. Of cause, the property offers the possibility to register multiple passthroughs at once, but a loop just adds one line.

Still, I could imagine the situation, where someone wants to check which passthroughs are already registered. which I addressed with the comment
> If you think there should be a way to query the registered Passthrough responses, I might implement something as read only property, but I currently do not see the necessity and would follow the good old practice "if in doubt leave it out".
Therefore, I could imagine to expose the registered passthroughs with a property (only the getter), which instead of the prefixes would provide the corresponding `PassthroughResponse` instances, which facilitates to use the response registry methods (remove, replace, upsert) with them.